New BAIP IAM Ref Arch (#1177)
SAP/architecture-center: docs/ref-arch/RA0019/5-SAP-manged-Joule-w-BusinessAIPlatform/readme.md

---
id: 3ec73b

> slug: /ref-arch/3ec73b

Reviewer: Could you change the id to 4c80fa?(also with in the slug)

The id must be unique and the previous one is already occupied by another document.

We need to communicate this somewhere. I will take a note

sidebar_position: 5
title: SAP Joule Landscape Recommendation
description: >-
  Recommended landscape setup for a unified SAP Joule experience
keywords:
  - sap
  - identity authentication
  - cloud identity
  - erp security solutions
  - access management
  - joule
  - joule studio
  - custom joule skills
  - ai agents
  - sap integration
  - sap ai
  - automation
  - hybrid landscapes
  - staged landscape
  - landscape
  - business ai platform
sidebar_label: Joule Landscape Recommendation
image: img/ac-soc-med.png
tags:
  - security
  - joule
  - sap-managed
  - cloud_identity

>   - sci

Reviewer: Some of the referenced tags don't exist in docs/tags.yml, which causes the build to fail. If you see a need for a new tag that would be relevant to more than one document, feel free to add one there.

Author: solved

hide_table_of_contents: false
hide_title: false
toc_min_heading_level: 2
toc_max_heading_level: 4
draft: false
unlisted: false
contributors:
  - sapgunnar
  - gunnar-kosche_sap
  - fabianleh
  - NormanNuernberger
discussion: 
last_update:
  author: NormanNuernberger
  date: 2025-07-08
---

> Setting up Joule Work in an existing SAP landscape requires an integration into SAP Cloud Identity Services. Joule and SAP Cloud Identity services are provided in two stages (Test and Production), while many other SAP systems in a typical landscape exist in more than 2 stages, such as Sandbox, Development, Test, Pre-Prod and Production. Identities in such landscapes with more than 2 stages are typically treated as productive also on non-productive systems. For example, a developer on a development system is a productive developer. To support such scenarios, as well as a strict SaaS approach (two-staged), SAP is providing two different architectural diagrams for landscape recommendations - one for a two-staged landscape (SaaS) and one for an n-staged landscape.

Reviewer 2: ```suggestion
Setting up Joule Work in an existing SAP landscape requires an integration into SAP Cloud Identity Services. Joule and SAP Cloud Identity Services are provided in two stages (Test and Production), while many other SAP systems in a typical landscape exist in more than 2 stages, such as Sandbox, Development, Test, Pre-Prod and Production. Identities in such landscapes with more than 2 stages are typically treated as productive also on non-productive systems. For example, a developer on a development system is a productive developer. To support such scenarios, as well as a strict SaaS approach (two-staged), SAP is providing two different architectural diagrams for landscape recommendations - one for a two-staged landscape (SaaS) and one for an n-staged landscape.
```

# Two-Staged Landscape Architecture

> The architecture diagram depicts the recommended SAP Joule setup for a two-staged Joule landscape, illustrating how the various components are organized across Test, and Production stages.

Reviewer: Please don't use an h1 markdown heading within the document content. Docusaurus takes the title from the frontmatter in the beginning of the document and takes care of rendering it as h1 once in the browser.

Otherwise it looks off

<img width="979" height="583" alt="Image" src="https://github.com/user-attachments/assets/0052cee7-1bf1-47cd-81c6-106c51978e6d" />

Author: updated

- **SAP Cloud Identity Services**: Test and Productive tenants of SAP Cloud Identity Services manage user authentication and authorization. The Test tenant serves the Test stage, while the Productive tenant serves the Production stage. Both integrate with the respective Corporate Identity Provider (Pre-Prod or Prod) for enterprise single sign-on.

- **SAP Business AI Platform**: SAP Business AI Platform is an enterprise AI foundation that brings together AI, data, process context, and governance, so organizations can build, deploy, integrate, manage and scale AI capabilities into business processes and applications.
The following services are hosted, SAP-managed, by the *SAP Business AI Platform*, separated by stage:
  - **Joule Work**: SAP’s digital assistant layer that brings conversational and generative AI into SAP applications and workflows.
  - **Joule Studio**: Low-code tooling to design, configure, and manage Joule-based experiences, including agents and skills.

>   - **SAP Agentic Layer**: Runtime and integrated management platform for AI agents.

Reviewer 2: Marked by brand check but I would be OK if you want to use anyway.

1. "SAP Agentic Layer" — not an official SAP product name (Medium) Used twice, as a sub-bullet under SAP Business AI Platform in both architecture sections. SAP does not ship a product called "SAP Agentic Layer." The official branded names for this capability are Joule Agents (autonomous AI workers) and Joule Assistants (orchestrators that coordinate agents). Consider replacing or restructuring the bullet to reflect these terms. [Source](https://www.sap.com/products/artificial-intelligence/ai-agents.html)

- **Centralized identity management**: SAP Cloud Identity Services tenants (Test and Productive) provide consistent authentication and authorization across all stages. Integration with Corporate Identity Providers ensures that enterprise security policies are enforced, while the test tenant enables safe validation of identity configurations before production deployment.

## Specialities

- **Signavio**: Signavio does not provide test tenants. Signavio will only be integrated into the Production stage.

- **Cloue Application Lifecycle Management (CALM)**: CALM does not provide test tenants. CALM will only be integrated into the Production stage.

# N-staged Landscape Architecture

The architecture diagram depicts the recommended SAP Joule setup for an n-staged Joule landscape, illustrating how the various components are organized across the stages. In this example a 3-staged landscape and a sandbox stage in addition.

> ![drawio](drawio/n_staged_BAIP_IAM_Architecture.drawio)

Reviewer: note that you could set a different title shown in the rendered svg of the diagram as show here (if you wish ofc)

https://architecture.learning.sap.com/docs/community/components#calling-the-drawio-component-in-the-page-body-of-the-readmemd-file

Author: skip it for now

The solution architecture consists of the following key elements:

> - **Three staged environments** (Development, Test and Production): Each stage is represented as a horizontal row in the diagram. All stages are sharing the same instance of SAP Cloud Identity Services and the SAP Business AI Platform. The Test instances of SAP Cloud Identity Services and SAP Business AI Platform are connected to a separated Sandbox landscape. This separation ensures that changes can be developed and validated before affecting production workloads, while identities in non-productive systems can be managed like productive identities.

Reviewer 2: ```suggestion
- **Three staged environments** (Development, Test and Production): Each stage is represented as a horizontal row in the diagram. All stages share the same instance of SAP Cloud Identity Services and the SAP Business AI Platform. The Test instances of SAP Cloud Identity Services and SAP Business AI Platform are connected to a separate Sandbox landscape. This separation ensures that changes can be developed and validated before affecting production workloads, while identities in non-productive systems can be managed like productive identities.
```

- **SAP Business AI Platform**: SAP Business AI Platform is an enterprise AI foundation that brings together AI, data, process context, and governance, so organizations can build, deploy, integrate, manage and scale AI capabilities into business processes and applications.
The following services are hosted, SAP-managed, by the *SAP Business AI Platform*, separated by stage:
  - **Joule Work**: SAP’s digital assistant layer that brings conversational and generative AI into SAP applications and workflows.
  - **Joule Studio**: Low-code tooling to design, configure, and manage Joule-based experiences, including agents and skills.

>   - **SAP Agentic Layer**: Runtime and integrated management platform for AI agents.

Reviewer 2: "SAP Agentic Layer" doesn't seem to be approved by SAP Branding, yet...

- **SAP Cloud Identity Services**: Test and Productive tenants of SAP Cloud Identity Services manage user authentication and authorization. The Test tenant serves a dedicated Sandbox stage only, while the Productive tenant serves all the other stages. Both integrate with the respective Corporate Identity Provider (Pre-Prod or Prod) for enterprise single sign-on.

- **SAP Cloud Solutions per stage**: Each stage connects to the corresponding instances of SAP cloud applications such as SAP S/4HANA Cloud, SAP SuccessFactors, and other Joule-enabled solutions. This ensures that Joule skills and configurations are tested against non-productive data before being deployed to production.

## Characteristics

- **Staged development lifecycle**: The n-staged model provides a structured path for developing, validating, and deploying Joule skills, agents, and configurations. Changes are developed in the Development stage and qualified in the Test stage before promotion to production, reducing the risk of disruptions to business operations, while developers and test users are treated like productive users to mitigate the risk of interruptions in the development lifecycles.

> - **Centralized identity management**: SAP Cloud Identity Services tenants (Test and Productive) provide consistent authentication and authorization across all stages. Integration with Corporate Identity Providers ensures that enterprise security policies are enforced, while the test tenant enables safe validation of identity configurations before production deployment.

Reviewer 2: "Corporate Identity Providers" - should this be in capital letters?

Author: not req in my pov

> ## Specialities

Reviewer 2: Section heading "Specialities" (used twice) — British spelling. Consider "Specialties" if you're targeting American English technical audiences. Either is fine as long as it's applied consistently.

> - **SAP SuccessFactors**: SAP SuccessFactors is owning an own identifier for a person, the Person UID, required for Embedded Analytics. This UID has an independent lifecycle than the Global User UID, which is owned by the SAP Cloud Identity Services. Connecting multiple instances of SAP SuccessFactors with a single instance of SAP Cloud Identity Services may lead to conflicts when synchronizing the identities from SAP SuccessFactors to SAP Cloud Identity Services. There are different possibilities to avoid such conflicts:

Reviewer 2: ```suggestion
- **SAP SuccessFactors**: SAP SuccessFactors owns an own identifier for a person, the Person UID, required for Embedded Analytics. This UID has an independent lifecycle than the Global User UID, which is owned by the SAP Cloud Identity Services. Connecting multiple instances of SAP SuccessFactors with a single instance of SAP Cloud Identity Services may lead to conflicts when synchronizing the identities from SAP SuccessFactors to SAP Cloud Identity Services. There are different possibilities to avoid such conflicts:
```

> - **SAP SuccessFactors**: SAP SuccessFactors is owning an own identifier for a person, the Person UID, required for Embedded Analytics. This UID has an independent lifecycle than the Global User UID, which is owned by the SAP Cloud Identity Services. Connecting multiple instances of SAP SuccessFactors with a single instance of SAP Cloud Identity Services may lead to conflicts when synchronizing the identities from SAP SuccessFactors to SAP Cloud Identity Services. There are different possibilities to avoid such conflicts:

Reviewer 2: ```suggestion
- **SAP SuccessFactors**: SAP SuccessFactors is owning an own identifier for a person, the Person UID, required for Embedded Analytics. The UID lifecycle is independent from the lifecycle of the Global User UID, which is owned by the SAP Cloud Identity Services. Connecting multiple instances of SAP SuccessFactors with a single instance of SAP Cloud Identity Services may lead to conflicts when synchronizing the identities from SAP SuccessFactors to SAP Cloud Identity Services. There are different possibilities to avoid such conflicts:
```

>     - By using different user accounts in different instances of SAP SuccessFactors, while authentication via Global User UID can allow single sign on even with different mail addresses in the different instances of SAP SuccessFactors.

Reviewer 2: ```suggestion
    - By using different user accounts in different instances of SAP SuccessFactors, while authentication via Global User UID can allow single sign-on even with different mail addresses in the different instances of SAP SuccessFactors.
```
